File: app/src/main/java/com/openmobl/pttDriver/service/IDeviceDriverService.java
package com.openmobl.pttDriver.service;

import android.bluetooth.BluetoothDevice;

import com.openmobl.pttDriver.model.Device;
import com.openmobl.pttDriver.model.PttDriver;

public interface IDeviceDriverService {
    void setPttDevice(Device device);
    boolean deviceIsValid();

    void setPttDriver(PttDriver driver);

    void setAutomaticallyReconnect(boolean autoReconnect);
    boolean getAutomaticallyReconnect();

    void registerStatusListener(DeviceStatusListener statusListener);
    void unregisterStatusListener(DeviceStatusListener statusListener);

    DeviceConnectionState getConnectionState();

    void connect();
    void disconnect();
}

File: app/src/main/java/com/openmobl/pttDriver/service/IDeviceDriverService.kt
package com.openmobl.pttDriver.service

import com.openmobl.pttDriver.model.Device
import com.openmobl.pttDriver.model.PttDriver

interface IDeviceDriverService {
    fun setPttDevice(device: Device?)
    fun deviceIsValid(): Boolean
    fun setPttDriver(driver: PttDriver?)
    var automaticallyReconnect: Boolean
    fun registerStatusListener(statusListener: DeviceStatusListener?)
    fun unregisterStatusListener(statusListener: DeviceStatusListener?)
    val connectionState: DeviceConnectionState
    fun connect()
    fun disconnect()
}